sdks/python/apache_beam/coders/coder_impl.py

```diff
@@ -73,6 +73,11 @@
 except ImportError:
   dataclasses = None  # type: ignore
 
+if sys.version_info < (3, 11):
```

Comment:
let's use user-specified pickler like in other places. the ux should allow user to override the default pickler in once place at start time

Reply:
Yes. True. My goal here is to see what tests fail and how to resolve the first.

Once they are resolved, I will takle on adding dill as an optional dependency